lib/Raven/Client.php

```diff
@@ -112,13 +113,26 @@ class Client
     protected $config;
 
     /**
+     * @var HttpAsyncClient The HTTP client
+     */
+    private $httpClient;
+
+    /**
+     * @var MessageFactory The message factory
+     */
+    private $messageFactory;
+
+    /**
      * Constructor.
      *
-     * @param Configuration $config The client configuration
+     * @param Configuration   $config     The client configuration
+     * @param HttpAsyncClient $httpClient The HTTP client
      */
-    public function __construct(Configuration $config)
+    public function __construct(Configuration $config, HttpAsyncClient $httpClient, MessageFactory $messageFactory)
```

Comment:
>I say you could use all params as separate properties in `Configuration`

The factories instances or the HTTP client instance are not options, so why should they be in that class? The `Client` class is indeed harder to instantiate because of the number of parameters it needs, this is why the builder was introduced. If you want to instantiate the client class directly then you will have to deal with the fact that it needs dependencies